### PR TITLE
include automated field when editing contacts

### LINKED ISSUE #2136: Editing contact removes server_set from extra data
Unlikely occurrence, but just happened to me :)

### PR DESCRIPTION
**Pull request by struan**
Display fields that are automatically filled in the contact editing form
otherwise the data is lost. Also include the automated field in the
form.

Fixes #2136

### CODE REVIEW

File: templates/web/base/admin/extra-metadata-form.html [hunk 1]
```diff
@@ -6,7 +6,13 @@
...
         <label>
             [% loc('Order') %]
-            <input name="metadata[[% loop.index %]].order" data-field-name="order" type=text value="[% meta.order | html %]">
+            <input name="metadata[[% loop.index %]].order" data-field-name="order" type=text value="[% meta.order OR 0 | html %]">
+        </label>
+
+        <div class="admin-hint"><p>[% loc('Whether the field is displayed to the user, in which case it will be blank, otherwise it will be automatically populated. This field is set usually set automatically.') %]</p></div>
+        <label>
+            [% loc('Automated') %]
+            <input name="metadata[[% loop.index %]].automated" data-field-name="automated" type=text value="[% meta.automated | html %]">
```
Comment (davea) [inline templates/web/base/admin/extra-metadata-form.html]: Would this be better as a `<select>` given the limited number of options?

Comment (struan) [reply templates/web/base/admin/extra-metadata-form.html]: Yeah, good point.

File: templates/web/base/admin/extra-metadata-form.html [hunk 2]
```diff
@@ -6,7 +6,18 @@
         <div class="admin-hint"><p>[% loc('The ordering of this field on the report page. Fields are shown in ascending order according to this value.') %]</p></div>
         <label>
             [% loc('Order') %]
-            <input name="metadata[[% loop.index %]].order" data-field-name="order" type=text value="[% meta.order | html %]">
+            <input name="metadata[[% loop.index %]].order" data-field-name="order" type=text value="[% meta.order OR 0 | html %]">
+        </label>
+
+        <div class="admin-hint"><p>[% loc('Whether the field is displayed to the user, in which case it will be blank, otherwise it will be automatically populated. This field is set usually set automatically.') %]</p></div>
```
Comment (dracos) [inline templates/web/base/admin/extra-metadata-form.html]: Typo here: "set usually set". This doesn't explain the hidden field/server set difference, perhaps something like "Whether the field is displayed to the user, output in the form as a hidden field to be populated automatically, or only generated by the server upon Open311 submission" or similar?

File: templates/web/base/admin/extra-metadata-form.html (hunk 1 above)
Comment (dracos) [reply templates/web/base/admin/extra-metadata-form.html]: I see the select has been added, but the `<input>` is still here unless I'm being dense :)

File: templates/web/base/admin/extra-metadata-form.html [hunk 3]
```diff
@@ -6,7 +6,18 @@
...
+        <label>
+            [% loc('Automated') %]
+            <input name="metadata[[% loop.index %]].automated" data-field-name="automated" type=text value="[% meta.automated | html %]">
+            <select name="metadata[[% loop.index %]].automated" data-field-name="automated" class="js-metadata-item-type">
+                <option value="" [% meta.automated == '' ? 'selected' : '' %]></option>
+                <option value="server_set" [% meta.automated == 'string' ? 'selected' : '' %]>[% loc('Server Set') %]</option>
```
Comment (dracos) [inline templates/web/base/admin/extra-metadata-form.html]: should be `server_set`, not `string`

**Pull request merged**